sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpressions.scala

```diff
@@ -114,22 +114,14 @@ abstract class GpuUnaryExpression extends UnaryExpression with GpuExpression {
   }
 
   override def columnarEval(batch: ColumnarBatch): Any = {
-    val input = child.columnarEval(batch)
-    try {
-      input match {
-        case vec: GpuColumnVector =>
-          doItColumnar(vec)
-        case other =>
-          withResource(GpuScalar.from(other, child.dataType)) { s =>
-            withResource(GpuColumnVector.from(s, batch.numRows(), child.dataType)) { vec =>
-              doItColumnar(vec)
-            }
+    withResourceIfAllowed(child.columnarEval(batch)) {
```

Comment:
Was this recommended by IntelliJ? If not then perhaps it makes sense to make this PR only restricted to auto changes so we don't have to spend much time on this

Reply:
it was flagged because of the use of isInstanceOf asInstanceOf which is an anti-pattern which should be handled by pattern-matching. I went an extra mile to rewrite it for what was actually meant. there.

This PR is too big, I'll start splitting it up.